projects/components/src/modal/default-modal-ref.ts

```diff
@@ -0,0 +1,32 @@
+import { Observable, Observer, ReplaySubject } from 'rxjs';
+import { PopoverRef } from '../popover/popover-ref';
+import { ModalRef } from './modal';
+
+export class DefaultModalRef extends ModalRef<unknown> {
+  public readonly closed$: Observable<unknown>;
+
+  private readonly closedObserver: Observer<unknown>;
+  private popoverRef?: PopoverRef;
+
+  public constructor() {
+    super();
+    const closedSubject = new ReplaySubject<unknown>(1);
+    this.closedObserver = closedSubject;
```

Comment:
Is there a reason to not just assign `closedObserver` directly rather than declaring `closedSubject` first?

Also, why name it `closedObserver` when its a subject?

Reply:
A subject is both an observable and an observer, so what I'm doing here is splitting it up into its component interfaces. If I directly assign it to closedObserver, I no longer can get the observable view of it. If I change closedObserver to be typed as a subject, I'm holding on to an implementation class that has more details than I want this class to know about.